src/engine/UIManager.cxx

```diff
@@ -901,6 +902,19 @@ void UIManager::initializeDollarVariables()
         combobox->setActiveID(Settings::instance().fullScreenMode);
         combobox->registerCallbackFunction(Signal::slot(this, &UIManager::changeFullScreenMode));
       }
+      else if (it->getUiElementData().elementID == "$MusicVolumeSlider")
+      {
+        Slider *slider = dynamic_cast<Slider *>(it.get());
+
+        if (!slider)
+        {
+          LOG(LOG_WARNING) << "Can't use element ID MusicVolumeSlider for an element other than a slider!";
+          continue;
+        }
+
+        slider->setValue(Settings::instance().musicVolume * 100);
+        slider->registerCallbackFunction(Signal::slot(this, &UIManager::changeMusicVolume));
```

Comment:
Consider following:
- Passing lambda instead of `UIManager::changeMusicVolume`
- Signal can actually give back current value instead of Slider object pointer. Could eliminate need for get function you made

Reply:
With lambda, I'm wondering if that would impact readability of the code. I just did a new function because I saw the other settings for display had their own functions, and I wanted to be consistent. I also don't know if maybe we would want to have music volume adjusted somewhere else in the future. I might be overthinking this, though. 

That being said, I'm not vehemently opposed, and I'm looking at how I would write a lambda for this.

Re: signal giving back current value, I didn't know that, that's much simpler!

Reply:
Regarding lambda, just try and see. If you do not like it I would respect, and we keep your approach.

For Slider::getValue story. I was thinking can you register callback callback which actually return slider value instead of slider element itself. Than you do not need to query for the value and use it directly for setting the audio volume.

Reply:
You're right about the current code not being state of the art. I need to do some refactoring for Slider and for this code as well, (including lambda function which I will do, now that I got some of my confusion cleared up in the discord). For the sake of moving forward and getting this feature in, I am going to merge it in just so it's there, and then I'll send you a DM on discord about these changes for refactoring since I do want to address these points.